OMP hangs when trying to exit the terminal

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

Everytime I try to exit of terminal  after execute any command OMP hangs, this is happenig since versions 29.22.x , versions of 29.21.x and before everything ok .

<img width="1920" height="980" alt="Image" src="https://github.com/user-attachments/assets/0032ca19-379e-4c03-88d4-a034024463e4" />



### Theme

atomic-mod.omp.json:

```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "alignment": "left",
      "segments": [
        {
          "background": "#0077c2",
          "foreground": "#ffffff",
          "leading_diamond": "\u256d\u2500\ue0b6",
          "style": "diamond",
          "template": "\uf120 {{ .Name }} ",
          "type": "shell"
        },
        {
          "background": "#ef5350",
          "foreground": "#FFFB38",
          "style": "diamond",
          "template": "<parentBackground>\ue0b0</> \uf292 ",
          "type": "root"
        },
        {
          "background": "#FF9248",
          "foreground": "#2d3436",
          "powerline_symbol": "\ue0b0",
          "options": {
            "folder_icon": " \uf07b ",
            "home_icon": "\ue617",
            "style": "folder"
          },
          "style": "powerline",
          "template": " \uf07b\u200b\uea9c\u200b {{ .Path }}",
          "type": "path"
        },
        {
          "background": "#FFFB38",
          "background_templates": [
            "{{ if or (.Working.Changed) (.Staging.Changed) }}#ffeb95{{ end }}",
            "{{ if and (gt .Ahead 0) (gt .Behind 0) }}#c5e478{{ end }}",
            "{{ if gt .Ahead 0 }}#C792EA{{ end }}",
            "{{ if gt .Behind 0 }}#C792EA{{ end }}"
          ],
          "foreground": "#011627",
          "powerline_symbol": "\ue0b0",
          "options": {
            "branch_icon": "\ue725 ",
            "fetch_status": true,
            "fetch_upstream_icon": true
          },
          "style": "powerline",
          "template": " {{ .UpstreamIcon }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }}<#ef5350> \uf046 {{ .Staging.String }}</>{{ end }} ",
          "type": "git"
        },
		{
		  "type": "winget",
		  "style": "powerline",
		  "powerline_symbol": "",
		  "foreground": "#ffffff",
		  "background": "#0077c2",
		  "template": "  {{ .UpdateCount }} ",
		  "cache": {
			"duration": "12h",
			"strategy": "device"
		  }
		},
        {
          "background": "#83769c",
          "foreground": "#ffffff",
          "options": {
            "style": "roundrock",
            "threshold": 0
          },
          "style": "diamond",
          "template": " \ueba2\u200b {{ .FormattedMs }}\u2800",
          "trailing_diamond": "\ue0b4",
          "type": "executiontime"
        }
      ],
      "type": "prompt"
    },	
    {
      "alignment": "right",
      "overflow": "hide",
      "segments": [
        {
          "background": "#306998",
          "foreground": "#FFE873",
          "leading_diamond": "\ue0b6",
          "style": "diamond",
          "template": "\ue235 {{ if .Error }}{{ .Error }}{{ else }}{{ if .Venv }}{{ .Venv }} {{ end }}{{ .Full }}{{ end }}",
          "trailing_diamond": "\ue0b4 ",
          "type": "python"
        },
		{
		  "type": "project",
		  "style": "diamond",
		  "trailing_diamond": "\ue0b4 ",
		  "leading_diamond": "\ue0b6",
		  "foreground": "#193549",
		  "background": "#ffeb3b",
		  "template": "{{ if .Error }}{{ .Error }}{{ else }}{{ if .Version }} {{.Version}}{{ end }}{{ if .Name }}{{ if eq .Type \"python\"}} {{ else }} {{ .Type }}{{ end }} {{ .Name }}{{ end }}{{ end }}"
		},
        {
          "background": "#0077C2",
          "foreground": "#ffffff",
          "leading_diamond": "\ue0b6",
          "options": {
            "linux": "\ue712",
            "macos": "\ue711",
            "windows": "\ue70f"
          },
          "style": "diamond",
          "template": " {{ if .WSL }}WSL at {{ end }}{{.Icon}} ",
          "type": "os"
        },
        {
          "background": "#f36943",
          "background_templates": [
            "{{if eq \"Charging\" .State.String}}#b8e994{{end}}",
            "{{if eq \"Discharging\" .State.String}}#fff34e{{end}}",
            "{{if eq \"Full\" .State.String}}#33DD2D{{end}}"
          ],
          "foreground": "#262626",
          "invert_powerline": true,
          "powerline_symbol": "\ue0b2",		  
          "options": {
            "charged_icon": "\uf240 ",
            "charging_icon": "\uf1e6 ",
            "discharging_icon": "\ue234 "
          },
          "style": "powerline",
          "template": " {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }}{{ .Error }}\uf295 ",
          "type": "battery"
        },
        {
          "background": "#0077D7",
          "foreground": "#ffffff",
          "invert_powerline": true,
          "powerline_symbol": "\ue0b2",
          "style": "powerline",
          "template": " {{ if eq .Type \"wifi\"}}\uf1eb{{ else if eq .Type \"ethernet\"}}\ueba9{{ end }} ",
          "type": "connection"
        },
        {
          "background": "#40c4ff",
          "foreground": "#ffffff",
          "invert_powerline": true,
          "options": {
            "time_format": "_2,15:04"
          },
          "style": "diamond",
          "template": " \uf073 {{ .CurrentDate | date .Format }} ",
          "trailing_diamond": "\ue0b4",
          "type": "time"
        }
      ],
      "type": "prompt"
    },
    {
      "alignment": "left",
      "newline": true,
      "segments": [
        {
          "foreground": "#21c7c7",
          "style": "plain",
          "template": "\u2570\u2500",
          "type": "text"
        },
        {
          "foreground": "#e0f8ff",
          "foreground_templates": ["{{ if gt .Code 0 }}#ef5350{{ end }}"],
          "options": {
            "always_enabled": true
          },
          "style": "plain",
          "template": "\ue285\ueab6 ",
          "type": "status"
        }
      ],
      "type": "prompt"
    }
  ],
  "tooltips_action": "prepend",
  "tooltips": [
    {
      "type": "ipify",
      "tips": [
        "sdx",
        "sdxt",
		"ip",
		"ssh",
		"scp",
		"sftp",
		"Test-Connection"
      ],
	  "style": "diamond",
	  "foreground": "#d6deeb",
	  "background": "#c386f1",
	  "leading_diamond": "",
	  "trailing_diamond": "",
	  "template": "{{ .IP }}",
	  "options": {
		"http_timeout": 1000,
		"cache_duration": "none"
	  }
	}	  
  ],  
  "version": 4,
  "transient_prompt": {
	"background": "transparent",
	"foreground": "#ffffff",
	"template": "\uf120 {{ .Shell }}> "
  },
  "upgrade": {
  "notice": true,
  "interval": "168h",
  "auto": false,
  "source": "cdn"
  },
  "streaming": 500
}
```


### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
oh-my-posh debug --plain

Version: 29.22.1

Shell: shell (7.6.3)

Prompt:

╭─ shell  ​​   2  ​ 0ms⠀                                                                          100      8,11:39 
╰─

Segments:

ConsoleTitle(false)                        -   0 ms
Shell(true)                                -   0 ms
Root(false)                                -   0 ms
Path(true)                                 -   2 ms
Git(false)                                 -   1 ms
Winget(true)                               -   1 ms
Executiontime(true)                        -   1 ms
Python(false)                              -   0 ms
Project(false)                             -   3 ms
Os(true)                                   -   1 ms
Battery(true)                              -   5 ms
Connection(true)                           -   4 ms
Time(true)                                 -   0 ms
Text(true)                                 -   0 ms
Status(true)                               -   0 ms

Run duration: 20.4602ms

Logs:

[DEBUG] 11:35:13.803 debug.go:39 → logging enabled, raw mode: false
[DEBUG] 11:35:13.813 store.go:]:189 → (session) key not found: prompt_count_cache
[TRACE] 11:35:13.813 store.go:](session prompt_count_cache) - 0s
[TRACE] 11:35:13.813 terminal.go:setPromptCount() - 0s
[TRACE] 11:35:13.813 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.813 terminal.go:setPwd:102 → C:\Users\myuser
[TRACE] 11:35:13.813 terminal.go:setPwd() - 0s
[TRACE] 11:35:13.813 terminal.go:Init() - 0s
[DEBUG] 11:35:13.813 path_windows.go:PackageFamilyName:44 → failed to get PackageFamilyName
[TRACE] 11:35:13.813 path_windows.go:PackageFamilyName() - 0s
[TRACE] 11:35:13.813 path.go:Path() - 0s
[DEBUG] 11:35:13.813 store.go:init:114 → (device) loading upgrade_check
[DEBUG] 11:35:13.813 store.go:init:114 → (device) loading DSC_CONFIGURATION
[DEBUG] 11:35:13.813 store.go:init:114 → (device) loading accent_color
[DEBUG] 11:35:13.813 store.go:init:114 → (device) loading segment_cache_Winget
[DEBUG] 11:35:13.813 store.go:init:114 → (device) loading INITVERSIONCMD
[DEBUG] 11:35:13.813 store.go:init:114 → (device) loading DSC_SHELL
[DEBUG] 11:35:13.813 store.go:init:114 → (device) loading INITVERSIONPWSH
[TRACE] 11:35:13.813 store.go:init(device omp.cache) - 670.1µs
[TRACE] 11:35:13.813 init.go:SessionID() - 0s
[TRACE] 11:35:13.813 path.go:Path() - 0s
[DEBUG] 11:35:13.813 store.go:init:114 → (session) loading CONFIG_SOURCE
[DEBUG] 11:35:13.813 store.go:init:114 → (session) loading CONFIG
[DEBUG] 11:35:13.813 store.go:init:114 → (session) loading toggle_cache
[TRACE] 11:35:13.813 store.go:init(session pwsh.ff69d02e-b2dc-4a37-9205-331324efb35b.omp.cache) - 0s
[DEBUG] 11:35:13.813 load.go:isTheme:386 → C:\Users\myuser\AppData\Local\Programs\oh-my-posh\themes\atomic-mod.omp.json is not a theme
[TRACE] 11:35:13.814 load.go:resolveConfigLocation() - 0s
[DEBUG] 11:35:13.814 store.go:]:202 → (device) found entry: DSC_CONFIGURATION - [0x27b24a4b0000 0x27b24a4b0180 0x27b24a4b0300]
[TRACE] 11:35:13.814 store.go:](device DSC_CONFIGURATION) - 0s
[DEBUG] 11:35:13.814 home.go:12 → C:\Users\myuser
[DEBUG] 11:35:13.814 resource.go:Add:42 → item already exists
[TRACE] 11:35:13.814 load.go:read() - 667.7µs
[DEBUG] 11:35:13.815 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.815 store.go:](session toggle_cache) - 111.8µs
[DEBUG] 11:35:13.815 store.go:]:225 → (session) setting entry: toggle_cache - map[] with duration: infinite
[TRACE] 11:35:13.815 store.go:](session toggle_cache) - 0s
[DEBUG] 11:35:13.815 store.go:]:225 → (device) setting entry: DSC_CONFIGURATION - [0x27b24a4b0000 0x27b24a4b0180 0x27b24a4b0300] with duration: infinite
[TRACE] 11:35:13.815 store.go:](device DSC_CONFIGURATION) - 0s
[TRACE] 11:35:13.815 load.go:Parse() - 1.3252ms
[TRACE] 11:35:13.815 load.go:Load() - 1.3252ms
[TRACE] 11:35:13.815 terminal.go:Shell() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.815 store.go:]:189 → (session) key not found: template_cache
[TRACE] 11:35:13.815 store.go:](session template_cache) - 0s
[TRACE] 11:35:13.815 cache.go:restoreCache() - 0s
[TRACE] 11:35:13.815 terminal_windows.go:Root() - 0s
[TRACE] 11:35:13.815 terminal.go:Shell() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:StatusCodes() - 0s
[TRACE] 11:35:13.815 terminal_windows.go:IsWsl() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.815 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.815 terminal_windows.go:IsWsl() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.815 terminal.go:User:236 → myuser
[TRACE] 11:35:13.815 terminal.go:User() - 0s
[DEBUG] 11:35:13.815 terminal.go:Host:253 → HP-x360-Pavilion-Laptop
[TRACE] 11:35:13.815 terminal.go:Host() - 0s
[TRACE] 11:35:13.815 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.815 terminal.go:Getenv:64 → NO DATA
[TRACE] 11:35:13.815 terminal.go:Getenv(SHLVL) - 0s
[DEBUG] 11:35:13.815 writer.go:Init:170 → terminal program: Windows Terminal
[DEBUG] 11:35:13.815 writer.go:Init:171 → terminal shell: shell
[DEBUG] 11:35:13.815 terminal.go:Getenv:64 → NO DATA
[TRACE] 11:35:13.815 terminal.go:Getenv(OMP_CACHE_DISABLED) - 0s
[DEBUG] 11:35:13.815 store.go:]:202 → (device) found entry: accent_color - 38;2;0;120;215|48;2;0;120;215
[TRACE] 11:35:13.815 store.go:](device accent_color) - 0s
[TRACE] 11:35:13.815 colors.go:SetAccentColor() - 0s
[TRACE] 11:35:13.815 terminal.go:Shell() - 0s
[DEBUG] 11:35:13.815 terminal.go:Getenv:64 → 7.6.3
[TRACE] 11:35:13.815 terminal.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 11:35:13.815 debug.go:PrintDebug:24 → segment: Title
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.815 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.815 terminal.go:GOOS() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.815 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.815 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.815 segment.go:Execute:178 → segment: Root
[DEBUG] 11:35:13.815 store.go:]:202 → (session) found entry: toggle_cache - map[]
[DEBUG] 11:35:13.815 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.815 store.go:](session toggle_cache) - 0s
[TRACE] 11:35:13.815 store.go:](session toggle_cache) - 0s
[DEBUG] 11:35:13.815 segment.go:isToggled:331 → no toggles found
[DEBUG] 11:35:13.815 segment.go:isToggled:331 → no toggles found
[TRACE] 11:35:13.815 terminal.go:StatusCodes() - 0s
[DEBUG] 11:35:13.815 map.go:debugf:85 → status_template: {{ .Code }}
[TRACE] 11:35:13.815 terminal_windows.go:Root() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.815 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.815 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.815 segment.go:Execute:178 → segment: Project
[DEBUG] 11:35:13.815 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.815 store.go:](session toggle_cache) - 0s
[DEBUG] 11:35:13.815 segment.go:isToggled:331 → no toggles found
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.815 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.815 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.815 text.go:patchTemplate:146 → {{ .Code }}
[DEBUG] 11:35:13.815 segment.go:Execute:178 → segment: Shell
[DEBUG] 11:35:13.815 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.815 store.go:](session toggle_cache) - 0s
[DEBUG] 11:35:13.815 segment.go:isToggled:331 → no toggles found
[DEBUG] 11:35:13.815 map.go:debugf:85 → mapped_shell_names: map[]
[TRACE] 11:35:13.815 terminal.go:Shell() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.815 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.815 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.815 segment.go:Execute:178 → segment: Connection
[DEBUG] 11:35:13.815 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.815 store.go:](session toggle_cache) - 0s
[DEBUG] 11:35:13.815 segment.go:isToggled:331 → no toggles found
[DEBUG] 11:35:13.815 map.go:debugf:85 → type: wifi|ethernet
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.815 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.815 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.815 map.go:debugf:85 → always_enabled: true
[DEBUG] 11:35:13.815 segment.go:Execute:178 → segment: Winget
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.815 store.go:]:202 → (session) found entry: toggle_cache - map[]
[DEBUG] 11:35:13.815 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.815 terminal.go:GOOS() - 0s
[TRACE] 11:35:13.815 store.go:](session toggle_cache) - 0s
[DEBUG] 11:35:13.815 segment.go:Execute:178 → segment: Text
[DEBUG] 11:35:13.815 segment.go:isToggled:331 → no toggles found
[DEBUG] 11:35:13.815 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.815 store.go:](session toggle_cache) - 0s
[TRACE] 11:35:13.815 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.815 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.815 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.815 segment.go:Execute:178 → segment: Os
[DEBUG] 11:35:13.815 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.816 store.go:](session toggle_cache) - 508.2µs
[DEBUG] 11:35:13.816 segment.go:isToggled:331 → no toggles found
[TRACE] 11:35:13.816 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.816 map.go:debugf:85 → windows: 
[TRACE] 11:35:13.816 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.816 segment.go:isToggled:331 → no toggles found
[TRACE] 11:35:13.816 terminal.go:Flags() - 0s
[TRACE] 11:35:13.816 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.816 store.go:]:202 → (device) found entry: segment_cache_Winget - {"Segment":{"Text":"  2 ","Index":4},"Updates":[{"Name":"Directory Opus","ID":"GPSoftware.DirectoryOpus","Current":"12.23   1","Available":"3.24      w"},{"Name":"PremiumSoft Navicat Premium 17.3","ID":"PremiumSoft.NavicatPremium","Current":"17.3.6  1","Available":"7.3.11    w"}],"UpdateCount":2}
[TRACE] 11:35:13.816 terminal.go:Flags() - 0s
[TRACE] 11:35:13.816 store.go:](device segment_cache_Winget) - 1.0775ms
[DEBUG] 11:35:13.816 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.816 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.816 segment.go:Execute:178 → segment: Executiontime
[DEBUG] 11:35:13.816 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.816 store.go:](session toggle_cache) - 0s
[DEBUG] 11:35:13.816 segment.go:isToggled:331 → no toggles found
[TRACE] 11:35:13.816 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.816 map.go:debugf:85 → always_enabled: false
[TRACE] 11:35:13.816 terminal.go:ExecutionTime() - 0s
[DEBUG] 11:35:13.816 map.go:debugf:85 → threshold: 0.000000
[DEBUG] 11:35:13.816 map.go:debugf:85 → style: roundrock
[TRACE] 11:35:13.816 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.816 home.go:12 → C:\Users\myuser
[DEBUG] 11:35:13.816 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.816 terminal.go:GOOS() - 0s
[TRACE] 11:35:13.816 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.816 segment.go:Execute:178 → segment: Time
[DEBUG] 11:35:13.816 segment.go:Execute:178 → segment: Path
[DEBUG] 11:35:13.816 store.go:]:202 → (session) found entry: toggle_cache - map[]
[DEBUG] 11:35:13.816 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.816 store.go:](session toggle_cache) - 0s
[DEBUG] 11:35:13.816 segment.go:isToggled:331 → no toggles found
[TRACE] 11:35:13.816 store.go:](session toggle_cache) - 0s
[DEBUG] 11:35:13.816 segment.go:restoreCache:368 → restored segment from cache:  Winget
[DEBUG] 11:35:13.816 segment.go:isToggled:331 → no toggles found
[DEBUG] 11:35:13.816 map.go:debugf:85 → display_cygpath: false
[TRACE] 11:35:13.816 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.816 map.go:debugf:85 → time_format: _2,15:04
[TRACE] 11:35:13.816 terminal.go:Flags() - 0s
[TRACE] 11:35:13.816 terminal.go:GOOS() - 0s
[TRACE] 11:35:13.816 terminal.go:Flags() - 0s
[TRACE] 11:35:13.816 terminal.go:Flags() - 0s
[TRACE] 11:35:13.816 terminal.go:Flags() - 0s
[TRACE] 11:35:13.816 separator.go:Separator() - 0s
[TRACE] 11:35:13.816 terminal.go:Shell() - 0s
[TRACE] 11:35:13.817 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.817 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.817 terminal.go:GOOS() - 0s
[TRACE] 11:35:13.817 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.817 home.go:12 → C:\Users\myuser
[DEBUG] 11:35:13.817 segment.go:Execute:178 → segment: Python
[TRACE] 11:35:13.817 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.817 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.817 store.go:](session toggle_cache) - 0s
[DEBUG] 11:35:13.817 segment.go:Execute:178 → segment: Battery
[DEBUG] 11:35:13.817 segment.go:isToggled:331 → no toggles found
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[DEBUG] 11:35:13.817 map.go:debugf:85 → display_mode: environment
[DEBUG] 11:35:13.817 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.817 store.go:](session toggle_cache) - 0s
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(package.json) - 1.5862ms
[DEBUG] 11:35:13.817 segment.go:isToggled:331 → no toggles found
[DEBUG] 11:35:13.817 map.go:debugf:85 → deno_files: [deno.json deno.jsonc]
[DEBUG] 11:35:13.817 map.go:debugf:85 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 11:35:13.817 map.go:debugf:85 → folders: [.venv venv virtualenv venv-win pyenv-win]
[DEBUG] 11:35:13.817 map.go:debugf:85 → home_enabled: false
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[DEBUG] 11:35:13.817 home.go:12 → C:\Users\myuser
[TRACE] 11:35:13.817 terminal_windows.go:IsWsl() - 0s
[TRACE] 11:35:13.817 terminal.go:Flags() - 0s
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(deno.json) - 0s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[DEBUG] 11:35:13.817 map.go:debugf:85 → mapped_locations_enabled: true
[TRACE] 11:35:13.817 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[DEBUG] 11:35:13.817 segment.go:Execute:178 → segment: Git
[DEBUG] 11:35:13.817 map.go:debugf:85 → windows_registry_icon: 
[DEBUG] 11:35:13.817 map.go:debugf:85 → windows_registry_icon: 
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[DEBUG] 11:35:13.817 store.go:]:202 → (session) found entry: toggle_cache - map[]
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(jsr.jsonc) - 0s
[TRACE] 11:35:13.817 store.go:](session toggle_cache) - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → cargo_files: [Cargo.toml]
[DEBUG] 11:35:13.817 home.go:12 → C:\Users\myuser
[DEBUG] 11:35:13.817 segment.go:isToggled:331 → no toggles found
[TRACE] 11:35:13.817 separator.go:Separator() - 0s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(Cargo.toml) - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → python_files: [pyproject.toml]
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(pyproject.toml) - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → mojo_files: [mojoproject.toml]
[DEBUG] 11:35:13.817 map.go:debugf:85 → disable_with_jj: false
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(mojoproject.toml) - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → php_files: [composer.json]
[TRACE] 11:35:13.817 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(composer.json) - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → home_icon: 
[DEBUG] 11:35:13.817 map.go:debugf:85 → dart_files: [pubspec.yaml]
[DEBUG] 11:35:13.817 map.go:debugf:85 → mapped_locations: map[]
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(pubspec.yaml) - 0s
[TRACE] 11:35:13.817 separator.go:Separator() - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → nuspec_files: [*.nuspec]
[TRACE] 11:35:13.817 terminal.go:GOOS() - 0s
[TRACE] 11:35:13.817 separator.go:Separator() - 0s
[TRACE] 11:35:13.817 terminal.go:GOOS() - 0s
[TRACE] 11:35:13.817 terminal.go:GOOS() - 0s
[TRACE] 11:35:13.817 terminal.go:GOOS() - 0s
[TRACE] 11:35:13.817 terminal.go:GOOS() - 0s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(*.nuspec) - 0s
[TRACE] 11:35:13.817 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → dotnet_files: [*.sln *.slnf *.slnx *.vbproj *.fsproj *.csproj]
[DEBUG] 11:35:13.817 map.go:debugf:85 → cycle: []
[DEBUG] 11:35:13.817 map.go:debugf:85 → folder_separator_template:
[DEBUG] 11:35:13.817 map.go:debugf:85 → folder_separator_icon: \
[DEBUG] 11:35:13.817 map.go:debugf:85 → cycle_folder_separator: false
[DEBUG] 11:35:13.817 map.go:debugf:85 → folder_format: %s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[DEBUG] 11:35:13.817 map.go:debugf:85 → edge_format: %s
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(*.sln) - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → left_format: %s
[DEBUG] 11:35:13.817 map.go:debugf:85 → right_format: %s
[TRACE] 11:35:13.817 terminal.go:Flags() - 0s
[TRACE] 11:35:13.817 terminal.go:GOOS() - 0s
[TRACE] 11:35:13.817 terminal.go:StackCount() - 0s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(*.slnf) - 0s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(*.slnx) - 0s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(*.vbproj) - 0s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(*.fsproj) - 0s
[DEBUG] 11:35:13.817 terminal.go:ResolveSymlink:195 → C:\Users\myuser
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(*.csproj) - 0s
[TRACE] 11:35:13.817 terminal.go:ResolveSymlink(C:\Users\myuser) - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → julia_files: [JuliaProject.toml Project.toml]
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(JuliaProject.toml) - 0s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(Project.toml) - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → lake_files: [lakefile.lean lakefile.toml]
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(lakefile.lean) - 0s
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(lakefile.toml) - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → powershell_files: [*.psd1]
[DEBUG] 11:35:13.817 terminal.go:HasFilesInDir:151 → false
[TRACE] 11:35:13.817 terminal.go:HasFilesInDir(*.psd1) - 0s
[DEBUG] 11:35:13.817 map.go:debugf:85 → always_enabled: false
[TRACE] 11:35:13.818 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.818 win32_windows.go:isWriteable:259 → not current user or in group
[DEBUG] 11:35:13.818 win32_windows.go:isWriteable:263 → current user is member of S-1-5-32-544
[DEBUG] 11:35:13.818 win32_windows.go:isWriteable:271 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 11:35:13.818 win32_windows.go:isWriteable:273 → user has write access
[TRACE] 11:35:13.818 terminal_windows.go:DirIsWritable() - 534.7µs
[TRACE] 11:35:13.818 terminal.go:Flags() - 0s
[ERROR] 11:35:13.818 terminal.go:HasParentFilePath:494 → GetFileAttributesEx .git: The system cannot find the file specified.
[TRACE] 11:35:13.818 terminal.go:HasParentFilePath(.git) - 534.7µs
[TRACE] 11:35:13.818 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.820 networks_windows.go:parseNetworkInterface:263 → Found wifi interface: XIAMi_10T_A12_Miu14
[TRACE] 11:35:13.820 networks_windows.go:wifiNetwork() - 1.1327ms
[TRACE] 11:35:13.820 terminal.go:Flags() - 0s
[TRACE] 11:35:13.822 terminal_windows_nix.go:BatteryState() - 5.4353ms
[DEBUG] 11:35:13.822 map.go:debugf:85 → charged_icon: 
[TRACE] 11:35:13.822 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.822 terminal.go:Getenv:64 → 3
[TRACE] 11:35:13.822 terminal.go:Getenv(POSH_CURSOR_LINE) - 0s
[DEBUG] 11:35:13.822 terminal.go:Getenv:64 → 1
[TRACE] 11:35:13.822 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 0s
[TRACE] 11:35:13.822 terminal.go:Flags() - 0s
[TRACE] 11:35:13.822 terminal.go:Flags() - 0s
[DEBUG] 11:35:13.822 text.go:patchTemplate:146 →  {{ .Data.Name }}
[DEBUG] 11:35:13.822 text.go:patchTemplate:146 →  ​​ {{ .Data.Path }}
[DEBUG] 11:35:13.823 text.go:patchTemplate:146 →   {{ .Data.UpdateCount }}
[DEBUG] 11:35:13.823 text.go:patchTemplate:146 →  ​ {{ .Data.FormattedMs }}⠀
[TRACE] 11:35:13.823 terminal.go:Shell() - 0s
[DEBUG] 11:35:13.823 text.go:patchTemplate:146 →  {{ if .WSL }}WSL at {{ end }}{{.Data.Icon}}
[DEBUG] 11:35:13.823 text.go:patchTemplate:146 →  {{ if not .Data.Error }}{{ .Data.Icon }}{{ .Data.Percentage }}{{ end }}{{ .Data.Error }}
[DEBUG] 11:35:13.823 text.go:patchTemplate:146 → {{if eq "Charging" .Data.State.String}}#b8e994{{end}}
[DEBUG] 11:35:13.823 text.go:patchTemplate:146 → {{if eq "Discharging" .Data.State.String}}#fff34e{{end}}
[DEBUG] 11:35:13.823 text.go:patchTemplate:146 → {{if eq "Full" .Data.State.String}}#33DD2D{{end}}
[DEBUG] 11:35:13.823 text.go:patchTemplate:146 →  {{ if eq .Data.Type "wifi"}}{{ else if eq .Data.Type "ethernet"}}{{ end }}
[DEBUG] 11:35:13.823 text.go:patchTemplate:146 →   {{ .Data.CurrentDate | date .Data.Format }}
[DEBUG] 11:35:13.824 terminal_windows.go:TerminalWidth:108 → terminal width: 120
[TRACE] 11:35:13.824 terminal_windows.go:TerminalWidth() - 921.3µs
[DEBUG] 11:35:13.824 engine.go:shouldFill:167 → no filler specified
[TRACE] 11:35:13.824 terminal.go:Shell() - 0s
[DEBUG] 11:35:13.824 text.go:patchTemplate:146 → {{ if gt .Code 0 }}#ef5350{{ end }}
[TRACE] 11:35:13.824 terminal.go:Flags() - 0s
[TRACE] 11:35:13.824 terminal.go:Shell() - 0s
[DEBUG] 11:35:13.824 store.go:]:225 → (session) setting entry: rprompt -  with duration: infinite
[TRACE] 11:35:13.824 store.go:](session rprompt) - 0s
[DEBUG] 11:35:13.824 store.go:]:225 → (session) setting entry: rprompt_length - 0 with duration: infinite
[TRACE] 11:35:13.824 store.go:](session rprompt_length) - 0s
[TRACE] 11:35:13.824 terminal.go:Shell() - 0s
[TRACE] 11:35:13.824 path.go:Path() - 0s
```

--- a/src/shell/scripts/omp.ps1
+++ b/src/shell/scripts/omp.ps1
@@ -146,6 +146,7 @@ New-Module -Name "oh-my-posh-core" -ScriptBlock {
     # globally for the OnIdle action to pick up.
     $global:_ompStreamingState = $script:Streaming
     $script:StreamingOnIdleJob = $null
+    $script:StreamingExitingJob = $null
 
     $env:POWERLINE_COMMAND = "oh-my-posh"
     $env:POSH_SHELL = "pwsh"
@@ -952,14 +953,67 @@ New-Module -Name "oh-my-posh-core" -ScriptBlock {
     function Enable-PoshStreaming {
         $global:_ompStreaming = $true
 
+        if (-not $script:ServeSupported) {
+            return
+        }
+
+        # A normal `exit` never runs the module's OnRemove handler, so nothing
+        # would tell the serve daemon to quit - and it only exits on stdin EOF,
+        # which requires this process to be gone. But pwsh's shutdown in turn
+        # waits for the reader runspace's pipeline thread, which is blocked on
+        # the daemon's stdout: a circular wait that hangs the terminal on exit.
+        # Break the cycle on PowerShell.Exiting: ask the daemon to quit (so it
+        # flushes its caches) and close its stdin - the EOF signal that works
+        # even if the quit line is lost - then kill it if it lingers. Its
+        # stdout then EOFs, the reader returns, and shutdown proceeds.
+        #
+        # Engine-event actions receive $null MessageData and lose module-scope
+        # closures, so state comes from $global:_ompStreamingState, like the
+        # OnIdle action.
+        if ($null -eq $script:StreamingExitingJob) {
+            $script:StreamingExitingJob = Register-EngineEvent -SourceIdentifier PowerShell.Exiting -Action {
+                $s = $global:_ompStreamingState
+
+                if ($null -eq $s) {
+                    return
+                }
+
+                if ($null -ne $s.ServeProcess -and -not $s.ServeProcess.HasExited) {
+                    try {
+                        $s.StdIn.WriteLine('{"command":"quit"}')
+                        $s.StdIn.Flush()
+                        $s.StdIn.Close()
+                    }
+                    catch {
+                    }
+
+                    if (-not $s.ServeProcess.WaitForExit(500)) {
+                        try {
+                            $s.ServeProcess.Kill()
+                        }
+                        catch {
+                        }
+                    }
+                }
+
+                # A lingering legacy per-prompt stream process exits by itself
+                # after its render, but don't let it outlive the session either.
+                if ($null -ne $s.Process -and -not $s.Process.HasExited) {
+                    try {
+                        $s.Process.Kill()
+                    }
+                    catch {
+                    }
+                }
+            }
+        }
+
         # Start the daemon during shell init rather than at the first prompt:
         # Process.Start() returns quickly and the spawn + engine warmup then
         # overlaps with the rest of the profile instead of delaying the first
         # prompt. Failure is fine - the first prompt retries and can still
         # fall back to the legacy per-prompt stream.
-        if ($script:ServeSupported) {
-            [void](Start-PoshServe)
-        }
+        [void](Start-PoshServe)
     }
 
     function Enable-PoshTooltips {
@@ -1187,6 +1241,15 @@ New-Module -Name "oh-my-posh-core" -ScriptBlock {
                     Unregister-Event -ErrorAction Ignore
                 Remove-Job $script:StreamingOnIdleJob -Force -ErrorAction Ignore
                 $script:StreamingOnIdleJob = $null
+            }
+
+            if ($null -ne $script:StreamingExitingJob) {
+                # only remove our own PowerShell.Exiting subscriber, other modules may have theirs
+                Get-EventSubscriber -SourceIdentifier PowerShell.Exiting -ErrorAction Ignore |
+                    Where-Object { $null -ne $_.Action -and $_.Action.InstanceId -eq $script:StreamingExitingJob.InstanceId } |
+                    Unregister-Event -ErrorAction Ignore
+                Remove-Job $script:StreamingExitingJob -Force -ErrorAction Ignore
+                $script:StreamingExitingJob = $null
             }
 
             Remove-Variable -Name _ompStreamingState -Scope Global -ErrorAction Ignore

